products/sentinel_one.py

```diff
@@ -77,72 +76,80 @@ class SentinelOne(Product):
     product: str = 's1'
-    creds_file: str  # path to credential configuration file
-    _limit: int = 1000 # Default limit set to PowerQuery's default of 1000.
-    _token: str  # AAD access token
-    _url: str  # URL of SentinelOne console
-    _site_id: Optional[str]  # Site ID for SentinelOne
-    _account_id: Optional[str]  # Account ID for SentinelOne
+    profile: str = 'default'
+    creds_file: Optional[str] = None # path to credential configuration file
+    _limit: int # Limit results
+    _token: Optional[str]  = None # AAD access token
+    _url: Optional[str] = None # URL of SentinelOne console
+    _account_names: Optional[list] = [] # Account Name(s) for SentinelOne
+    _account_ids: list() = [] # Account ID(s) for SentinelOne
+    _site_ids: list() = [] # Site ID(s) for SentinelOne
     _session: requests.Session
-    _queries: dict[Tag, list[Query]]
-    _last_request: float
-    _site_ids: list[str]
-    _query_base: Optional[str]
-    _pq: bool  # Run queries using PowerQuery instead of DeepVisibility
-
-    def __init__(self, profile: str, creds_file: str, account_id: Optional[list[str]] = None,
-                 site_id: Optional[list[str]] = None, account_name: Optional[list[str]] = None, pq: bool = False,
-                 **kwargs):
-        if not os.path.isfile(creds_file):
-            raise ValueError(f'Credential file {creds_file} does not exist')
-
-        self.creds_file = creds_file
-        self._queries = dict()
-        self._query_base = None
-        self._pq = pq
-
-        # If no conditions match, the default limit will be set to PowerQuery's default of 1000.
-        if self._pq and self._limit >= int(kwargs.get('limit',0)) > 0:
-            self._limit = int(kwargs['limit'])
-
-        elif not self._pq and 20000 > int(kwargs.get('limit',0)) > 0:
-                self._limit = int(kwargs['limit'])
-
-        elif not self._pq: 
-            self._limit = 20000
+    _queries: dict[Tag, list[Query]] = dict()
+    _last_request: float = 0.0
+    _query_base: Optional[str] = None
+    _pq: bool # Run queries using PowerQuery instead of Deep Visibility
+    _raw: bool = False
+
+    def __init__(self, pq: bool = False, **kwargs):
+  
+        self.profile = kwargs['profile'] if 'profile' in kwargs else 'default'
+        self._site_ids = kwargs['site_ids'] if 'site_ids' in kwargs else []
+        self._account_ids = kwargs['account_ids'] if 'account_ids' in kwargs else []
+        self._account_names = kwargs['account_names'] if 'account_names' in kwargs else []
+        self._url = kwargs['url'] if 'url' in kwargs else None
+        self._token = kwargs['token'] if 'token' in kwargs else None
+        self.creds_file = kwargs['creds_file'] if 'creds_file' in kwargs else None
+        self._raw = kwargs['raw'] if 'raw' in kwargs else self._raw
+        limit = (kwargs['limit']) if 'limit' in kwargs else 0
+        self._pq = pq # This supports command-line options, will default to Deep Visibility
```

Comment:
This could be considered a breaking change making DV the default instead of PQ

In PR #94 , there was the design decision to make PQ the default due to accuracy and performance

Reply:
PQ should remain the default as DV will be sunsetting in the next 6 months to a year.

Reply:
That comment has since been corrected. Thanks! @xC0uNt3r7hr34t